### ISSUE TITLE
Fix invalid file status for disabled or deleted add-ons

### ISSUE THREAD

**Issue by wagnerand** [mozilla/addons-server#13795]
Some deleted or disabled add-ons have files that are approved, preliminary approved or even pending review. These files should be disabled for consistency and to ensure manual sql queries produce expected results.

@diox mentioned we should split it up in chunks to reduce load.
I would appreciate if we could do it by running an sql `UPDATE` statement manually against the database to keep effort low and lead time short.

There are two proposals so far to do this:

```
update files set status = 5
where id in
(select f.id from addons a
join versions v on a.id = v.addon_id
join files f on v.id = f.version_id
where a.status IN (5,11) AND MOD(f.id, 6) = 0
group by f.id);
```

or

```
update files set status = 5
where id in
(
select f.id from addons a
join versions v on a.id = v.addon_id
join (select id, version_id from files where status <> 5 order by id limit 50000) f on f.version_id = v.id
where a.status IN (5,11)
group by f.id
order by f.id
)
LIMIT 50000;
```

Either of those would have to be run manually several times until no more rows get updated.

**Comment by jvillalobos** [comment]
For the first proposal, the followup queries would have `MOD(f.id, 6) = 1`, `MOD(f.id, 6) = 2`, ..., `MOD(f.id, 6) = 5`, in case that isn't clear.

**Comment by diox** [comment]
First one gives me `ERROR 1093 (HY000): You can't specify target table 'files' for update in FROM clause`.

There is a `filesort` and `temporary table` that would be better to avoid in the second one, but we can probably live with it. I like the second one also because we can adjust the limit if we notice it's going well.

**Comment by wagnerand** [comment]
@diox I didn't manage to avoid the temporary table. Resolving that subquery makes the statement fail because mysql does not support querying the table column you are trying to update.

**Comment by wagnerand** [comment]
The filesort we could probably avoid, I'm not sure we actually gain much from it. We'd have to run the query several times anyway, also more times than `total amount / files per chunk` since not all files we fetch in the subquery might belong to an add-on that is disabled or deleted.

**Comment by diox** [comment]
Yeah, we'll do one run with the (second) query as-is on prod and see how that goes.

**Comment by bqbn** [comment]
We ran the second command 5 times in -prod, and here are the results for each run.

```
Query OK, 22831 rows affected (8.91 sec)
Rows matched: 22831  Changed: 22831  Warnings: 0


Query OK, 5228 rows affected (8.06 sec)
Rows matched: 5228  Changed: 5228  Warnings: 0


Query OK, 870 rows affected (8.08 sec)
Rows matched: 870  Changed: 870  Warnings: 0


Query OK, 162 rows affected (8.01 sec)
Rows matched: 162  Changed: 162  Warnings: 0


Query OK, 14 rows affected (8.16 sec)
Rows matched: 14  Changed: 14  Warnings: 0
```

As a result, the 6th run has no row to update.

**Comment by wagnerand** [comment]
Hm, even before we ran the statements, it seems the `hide_disabled_files` and `unhide_disabled_files` cronjobs are already overlapping.

2020-03-20 17:25:07.219 CET: Beginning job: hide_disabled_files
2020-03-20 17:55:07.072 CET: Beginning job: unhide_disabled_files
2020-03-20 18:00:45.608 CET: Ending job: hide_disabled_files
2020-03-20 18:02:48.788 CET: Ending job: unhide_disabled_files
(2020-03-20 18:10 CET - 18:24 CET: Queries above were run)

@diox is this a concern?

**Comment by diox** [comment]
With the way the crons are written nowadays, it shouldn't be a problem if `hide_disabled_files` and `unhide_disabled_files` run in parallel, because they look at different rows (they use the same base query rules but one is using them to filter, the other to exclude). 

Having the same cronjob overlapping is an issue, though.

**Comment by wagnerand** [comment]
https://github.com/mozilla/addons-server/issues/13797 should give us some answers.

**Comment by wagnerand** [comment]
The updated query to cover all relevant files is:
```
update files set status = 5
where id in
(
select f.id from addons a
join versions v on a.id = v.addon_id
join (select id, version_id from files where status <> 5 order by id) f on f.version_id = v.id
where a.status IN (5,11)
group by f.id
order by f.id
)
LIMIT 50000;
```

(`LIMIT` of the subquery removed)

**Comment by bqbn** [comment]
This is done. Last run updated 22K rows.

```
Query OK, 22242 rows affected (35.51 sec)
Rows matched: 22242  Changed: 22242  Warnings: 0
```

**Comment by wagnerand** [comment]
Thank you, @bqbn!
